src/main/kotlin/com/hibob/academy/resource/petsResource.kt

```diff
@@ -17,26 +18,31 @@ class PetsResource {
     @Consumes(MediaType.APPLICATION_JSON)
     fun createPet(@RequestBody pets: Pets): Response {
         Response.status(Response.Status.OK).build()
-        return Response.ok("POST OK").build()
+        return Response.ok(pets).build()
     }
 
     @GET
-    @Path("/{petId}/type")
+    @Path("/getPetType/{petId}")
     fun getPetType(@PathParam("petId") id: String): Response {
         Response.status(Response.Status.OK).build()
-        return Response.ok("GET OK").build()
+        return Response.ok("lab").build()
     }
 
     @PUT
-    @Path("/{petId}/type")
+    @Path("/updateType/{petId}")
     @Consumes(MediaType.APPLICATION_JSON)
-    fun updatePetType(@PathParam("petId") id: String, petType: String): Response {
+    fun updatePetType(@PathParam("petId") id: String, @QueryParam("newType") newType: String?): Response {
+        val pet = Pets(
+            name = "dog", type = "lab",
+            company_id = UUID.randomUUID(), date_of_arrival = Date()
+        )
+        pet.type = newType ?: pet.type
         Response.status(Response.Status.OK).build()
-        return Response.ok("PUT OK").build()
+        return Response.ok(pet).build()
     }
 
     @DELETE
-    @Path("/{petId}/type")
+    @Path("/deletePetType/{petId}/type")
```

Comment:
do you intend to delete a pet or a pet type here?

Reply:
so not sure I understand - you are getting a specific Id of a pet, and remove it's type and leave it without a type or all instances of the same type? anyways, I don't fully understand